lua/nvim-treesitter/indent.lua

```diff
@@ -4,50 +4,31 @@ local tsutils = require "nvim-treesitter.ts_utils"
 
 local M = {}
 
--- TODO(kiyan): move this in tsutils and document it
-local function get_node_at_line(root, lnum)
-  for node in root:iter_children() do
-    local srow, _, erow = node:range()
-    if srow == lnum then
-      return node
-    end
-
-    if node:child_count() > 0 and srow < lnum and lnum <= erow then
-      return get_node_at_line(node, lnum)
-    end
-  end
-end
-
-local function node_fmt(node)
-  if not node then
-    return nil
-  end
-  return tostring(node)
-end
-
 local get_indents = tsutils.memoize_by_buf_tick(function(bufnr, root, lang)
   local get_map = function(capture)
     local matches = queries.get_capture_matches(bufnr, capture, "indents", root, lang) or {}
     local map = {}
     for _, node in ipairs(matches) do
-      map[tostring(node)] = true
+      map[node:id()] = true
     end
     return map
   end
 
   return {
+    autos = get_map "@auto.node",
     indents = get_map "@indent.node",
+    dedents = get_map "@dedent.node",
     branches = get_map "@branch.node",
-    returns = get_map "@return.node",
     ignores = get_map "@ignore.node",
   }
 end, {
   -- Memoize by bufnr and lang together.
-  key = function(bufnr, _, lang)
-    return tostring(bufnr) .. "_" .. lang
+  key = function(bufnr, root, lang)
+    return tostring(bufnr) .. root:id() .. "_" .. lang
```

Comment:
This is needed in case the same language is injected onto itself (happens in rust).